lib/ansible/modules/network/files/net_put.py

```diff
@@ -42,6 +42,15 @@
       - Filename from src and at default directory of user shell on
         network_os.
     required: no
+  mode:
```

Comment:
Is it possible to guess the file type instead of taking it as input from the user?

Reply:
Not sure . We can do something like this - try to open it and if found some unicode error Exception then use binary mode. This approach might be heavily dependant on OS and environment so I am little hesitant to do it this way.  With current approach we will not try to open it unless told otherwise (mode = template) . It will work as ansible way of scp in most cases.

Reply:
@ganeshrn I noticed  template routine is not failing on my environment even if file has non ascii chars but failed on submitter.  (might be difference in python module lever version) For me it failed when trying file.write so auto-detect of file type might seem a bit unstable and env dependant . I think we might be better off in not trying to open it and transfer it unless customer is sure to put some vars and tell us through new option.